pkg/cluster/internal/providers/podman/provision.go

```diff
@@ -251,28 +252,67 @@ func getProxyEnv(cfg *config.Cluster, networkName string) (map[string]string, er
 			return nil, err
 		}
 		noProxyList := append(subnets, envs[common.NOProxy])
-		// Add pod and service dns names to no_proxy to allow in cluster
+		// Add pod, service and control plane(API server) dns names to no_proxy to allow in cluster
 		// Note: this is best effort based on the default CoreDNS spec
 		// https://github.com/kubernetes/dns/blob/master/docs/specification.md
 		// Any user created pod/service hostnames, namespaces, custom DNS services
 		// are expected to be no-proxied by the user explicitly.
-		noProxyList = append(noProxyList, ".svc", ".svc.cluster", ".svc.cluster.local")
+		noProxyList = append(noProxyList, ".svc", ".svc.cluster", ".svc.cluster.local", strings.Join([]string{cfg.Name, "control-plane"}, "-"))
```

Comment:
Changes to this method seem unrelated to netavark support and should also be kept consistent with the docker implementation

Reply:
This should be a separate commit or PR, and it should probably not reimplement naming the control plane endpoint with different logic.

Reply:
@BenTheElder : agreed, this is a separate problem. Should I create a separate issue for that and take this part out from the current PR?

Reply:
> This should be a separate commit or PR, and it should probably not reimplement naming the control plane endpoint with different logic.

Agreed. I've taken it out and will create a separate issue + another PR as an attempt to tackle the no_proxy inconsistencies. Thanks!

Reply:
thank you!